docs/intro/tutorial.rst

```diff
@@ -0,0 +1,213 @@
+
+Let's create our first Page Object abstraction by moving extraction logic
+outside our spider file.
+
+.. code-block:: python
+
+    from web_poet.pages import ItemWebPage
+
+
+    class BookPage(ItemWebPage):
+        """Extracts data from a book page"""
+
+        def to_item(self):
+            """Converts page into an item"""
+            return {
+                'url': self.url,
+                'name': self.css("title::text").get(),
+            }
+
+Now we have a ``BookPage`` class that implements the ``to_item`` method.
+We could go even further and extract a property from the ``to_item`` method.
+
+.. code-block:: python
+
+    from web_poet.pages import ItemWebPage
+
+
+    class BookPage(ItemWebPage):
+        """Extracts data from a book page"""
+
+        @property
+        def title(self):
+            """Extracts title from book page"""
+            return self.css("title::text").get()
+
+        def to_item(self):
+            """Converts page into an item"""
+            return {
+                'url': self.url,
+                'name': self.title,
+            }
+
+Changing spider
+===============
+
+The next step is to change our ``parse_book`` method in order to consume our
+newly created Page Object class.
+
+.. code-block:: python
+
+    def parse_book(self, response, book_page: BookPage):
+        """Extract data from book pages"""
+        yield book_page.to_item()
+
+The parser method now receives a type annotated argument called ``book_page``.
+Our Injection Middleware will detect it and provide the required dependencies.
+
+The spider should be looking like this:
+
+.. code-block:: python
+
+    class BooksSpider(scrapy.Spider):
+        """Crawl and extract books data"""
+
+        name = 'books'
+        start_urls = ['http://books.toscrape.com/']
+
+        def parse(self, response):
+            """Discovers book links and follows them"""
+            links = response.css('.image_container a')
+            yield from response.follow_all(links, self.parse_book)
+
+        def parse_book(self, response, book_page: BookPage):
+            """Extract data from book pages"""
+            yield book_page.to_item()
+
+You might have noticed that our parser method is quite simples and it's just
+returning the result of the ``to_item`` method call. We could make use of the
+``callback_for`` helper to reduce source code here.
+
+.. code-block:: python
+
+    from scrapy_poet import callback_for
+
+
+    class BooksSpider(scrapy.Spider):
+        """Crawl and extract books data"""
+
+        name = 'books'
+        start_urls = ['http://books.toscrape.com/']
+        parse_book = callback_for(BookPage)
+
+        def parse(self, response):
+            """Discovers book links and follows them"""
+            links = response.css('.image_container a')
+            yield from response.follow_all(links, self.parse_book)
+
+This helper could be used as an inline callback, but it would not work with
```

Comment:
I wonder if this can be reworded; "inline callback" can reming people of scrapy-inline-callbacks or twisted's `@inlineCallbacks`, but this is not what is meant here.

Reply:
Maybe just show an example: "you can also write `Request(url, callback_for(BookPage))` directly, without creating an attribute, but currently this won't work with Scrapy disk queues", something like that?

Reply:
I've just rewritten this text and formatted it as a note.